plugin.video.southpark_unofficial/southpark.py

```diff
@@ -0,0 +1,596 @@
+		ep_numb  = episode['episodeNumber'][2] + episode['episodeNumber'][3] ## EPISODE NUMBER
+		ep_aird  = episode['originalAirDate']
+
+		if episode['_availability'] == "banned":
+			ep_title += " [Banned]"
+			ep_mode   = PLUGIN_MODE_BANNED
+		elif episode['_availability'] == "beforepremiere":
+			ep_title += " [Premiere]"
+			ep_mode   = PLUGIN_MODE_PREMIERE
+			ep_desc   = "Premiere in {0}\n{1}".format(_premier_timeout(episode['originalAirDate']), ep_desc)
+		else:
+			ep_url = episode['itemId']
+
+		self.add_entry(ep_title, ep_url, ep_mode, ep_image, ep_desc, ep_seas, ep_numb, ep_aird, is_playable=True)
+
+	def add_directory(self, name, url, mode, iconimage="DefaultFolder.png"):
+		u = self.argv[0]+"?url="+urllib.quote_plus(url)+"&mode="+str(mode)
+		ok = True
+		liz = xbmcgui.ListItem(name, iconImage=iconimage, thumbnailImage=iconimage)
+		liz.setInfo(type="Video", infoLabels={"Title": name})
+		liz.setProperty("fanart_image", self.paths.DEFAULT_FANART)
+		ok = xbmcplugin.addDirectoryItem(handle=self.phandle, url=u, listitem=liz, isFolder=True)
+		return ok
+
+	def add_entry(self, name, url, mode, iconimage, desc="", season="", episode="", date="", is_playable=False):
+		name = _encode(name)
+		desc = _encode(desc)
+		if "?" in iconimage:
+			pos = iconimage.index('?') - len(iconimage)
+			iconimage = iconimage[:pos]
+		url       = "{0}?url={1}&mode={2}&title={3}&thumbnail={4}".format(self.argv[0], urllib.quote_plus(url), mode, name, iconimage)
+		convdate  = _date(date)
+		is_folder = not is_playable
+		entry = xbmcgui.ListItem(name, thumbnailImage=iconimage)
+		entry.setInfo(type="Video", infoLabels={"Title": name, "Plot": desc, "Season": season, "Episode": episode, "Aired": convdate})
+		entry.setProperty("fanart_image", self.paths.DEFAULT_FANART)
+		entry.setProperty("isPlayable", "true" if is_playable else "false")
+		xbmcplugin.setContent(self.phandle, 'episodes')
+		return xbmcplugin.addDirectoryItem(handle=self.phandle, url=url, listitem=entry, isFolder=is_folder)
+
+	def handle(self):
+		kodi = KodiParams(self.argv[2])
+		kodi.debug()
+		self.options.debug()
+		if kodi.PARAM_MODE == PLUGIN_MODE_SEASON:
+			self.create_episodes(kodi.PARAM_URL)
+		elif kodi.PARAM_MODE == PLUGIN_MODE_PLAY_EP:
+			self.play_episode(kodi.PARAM_URL, kodi.PARAM_EP_TITLE, kodi.PARAM_EP_THUMBNAIL)
+		elif kodi.PARAM_MODE == PLUGIN_MODE_FEATURED:
+			self.create_featured()
+		elif kodi.PARAM_MODE == PLUGIN_MODE_RANDOM:
+			self.create_random()
+		elif kodi.PARAM_MODE == PLUGIN_MODE_SEARCH:
+			self.create_search()
+		elif kodi.PARAM_MODE == PLUGIN_MODE_BANNED:
+			self.notify(self.i18n.WARNING_BANNED_EPISODE, WARNING_TIMEOUT_LONG)
+		elif kodi.PARAM_MODE == PLUGIN_MODE_PREMIERE:
+			self.notify(self.i18n.WARNING_PREMIERE, WARNING_TIMEOUT_LONG)
+		elif kodi.PARAM_MODE == '':
+			self.create_menu()
+		else:
+			self.notify("---ERROR---")
+			return
+		xbmcplugin.endOfDirectory(self.phandle)
+		xbmc.executebuiltin('Container.SetViewMode(504)')
```

Comment:
Setting the view mode isn't allowed for plugins.

Reply:
fixed in new pushed commit

Reply:
removed